contracts/periphery/src/execute.rs

```diff
@@ -1,125 +1,305 @@
-use cosmwasm_std::{attr, coin, to_binary, BankMsg, Env, MessageInfo, SubMsg, Uint128, WasmMsg};
+use cosmwasm_std::{
+    attr, coin, to_json_binary, BankMsg, Coin, CosmosMsg, Env, MessageInfo, Uint128, WasmMsg,
+};
 use cosmwasm_std::{DepsMut, Response};
-use ibcx_interface::periphery::{ExecuteMsg, SwapInfo};
+use ibcx_interface::periphery::{extract_pool_ids, ExecuteMsg, SwapInfo};
 use ibcx_interface::{core, helpers::IbcCore};
+use ibcx_pool::{query_pools, Simulator};
 
-use crate::state::{Context, CONTEXT};
-use crate::REPLY_ID_BURN_EXACT_AMOUNT_IN;
-use crate::{error::ContractError, msgs::make_mint_swap_exact_out_msgs};
+use crate::error::ContractError;
+use crate::{coin_sorter, deduct_fee, expand_fee, make_unit_converter};
 
-pub fn mint_exact_amount_out(
+pub fn mint_exact_amount_in(
     deps: DepsMut,
     env: Env,
     info: MessageInfo,
     core_addr: String,
-    output_amount: Uint128,
-    input_asset: String,
+    desired_denom: String,
+    min_index_amount: Uint128,
```

Comment:
This changes might effect to request payload, please check changes and notice to dongwu @byeongsu-hong

Reply:
I think it's already applied on front-end side. Isn't it? @dongwu-kim

Reply:
If you look below, it looks like the `mint_exact_amount_out` function still exists. 
And it's request payload is like below.
```
pub fn mint_exact_amount_out(
    deps: DepsMut,
    env: Env,
    info: MessageInfo,
    core_addr: String,
    index_amount: Uint128,
    input_denom: String,
    swap_info: Vec<SwapInfo>,
)
```
So the types of request payload was not changed.

Reply:
Then, the `mint_exact_amount_in` function is newly generated? @byeongsu-hong

Reply:
@hashableric yes.